### PR TITLE
Improve check_delegator_restriction function

### LINKED ISSUE #2508: Improve method check_delegator_restriction in delegation_based_restriction_mixin.py
First improvement:
Set the call parameter `lock_result` of the resulting db-calls to `False`. Reason: The result of the method are part of user permissions. It is not necessary to lock database in this cases.

Second improvement:
Reduce the database actions:

1. Read the meeting_user instances:
            meeting_users = self.datastore.filter(
                "meeting_user",
                And(
                    Or(
                        FilterOperator("meeting_id", "=", meeting_id)
                        for meeting_id in meeting_ids
                    ),
                    FilterOperator("user_id", "=", self.user_id),
                    FilterOperator("vote_delegated_to_id", "!=", None), 
                ),
                ["meeting_id"],
            )
2. If there are meeting_users in the above result set, get their meeting_ids
3. Read the meetings from point 2 with a get_many-db-call
4. Get the broken_meetings the way the old implementation does without the meeting_user-related condition

### PR DESCRIPTION
**Pull request by luisa-beerboom**
Closes #2508

### CODE REVIEW

File: openslides_backend/action/mixins/delegation_based_restriction_mixin.py [hunk 1]
```diff
@@ -51,43 +51,38 @@ def check_delegator_restriction(
...
-                        for meeting_id in meeting_ids
-                    ),
-                    Or(
-                        FilterOperator("id", "=", meeting_user_id)
-                        for meeting_user_id in operator["meeting_user_ids"]
+        if len(meeting_users):
+            delegation_meeting_ids = [
+                meeting_user["meeting_id"] for meeting_user in meeting_users.values()
+            ]
+            data = self.datastore.get_many(
```
Comment (hjanott) [inline openslides_backend/action/mixins/delegation_based_restriction_mixin.py:72]: I suggest calling this variable `delegation_meetings` or `delegation_meetings_data`.

File: openslides_backend/action/mixins/delegation_based_restriction_mixin.py [hunk 2]
```diff
@@ -51,43 +51,38 @@ def check_delegator_restriction(
...
+                lock_result=False,
             )
             broken_meetings: list[int] = []
-            for meeting_user in meeting_users.values():
-                meeting_id = meeting_user["meeting_id"]
-                if (
-                    (meeting := data["meeting"][meeting_id]).get(restriction)
-                    and (meeting.get("users_enable_vote_delegations"))
-                    and meeting_user.get("vote_delegated_to_id")
+            for meeting_id in delegation_meeting_ids:
```
Comment (hjanott) [inline openslides_backend/action/mixins/delegation_based_restriction_mixin.py:83]: I think we could make it more efficient using something such as:
`for meeting in data["meeting"].values():
        if meeting.get("restriction") and (`

File: openslides_backend/action/mixins/delegation_based_restriction_mixin.py (hunk 1 above)
Comment (luisa-beerboom) [reply openslides_backend/action/mixins/delegation_based_restriction_mixin.py]: Done

File: openslides_backend/action/mixins/delegation_based_restriction_mixin.py (hunk 2 above)
Comment (luisa-beerboom) [reply openslides_backend/action/mixins/delegation_based_restriction_mixin.py]: Done

**Review by hjanott — APPROVED** [review]

**Pull request merged**